contracts/examples/digital-cash/src/digital_cash.rs

```diff
@@ -68,15 +78,35 @@ pub trait DigitalCash {
             deposit.expiration_round >= self.blockchain().get_block_round(),
             "deposit expired"
         );
+
+        const HASH_DATA_BUFFER_LEN: usize = 1024;
+
+        let sig_len = signature.len();
         require!(
-            self.crypto().verify_ed25519(
-                &address.to_byte_array()[..],
-                &caller_address.to_byte_array()[..],
-                signature.to_boxed_bytes().as_slice()
-            ),
-            "invalid signature"
+            sig_len <= HASH_DATA_BUFFER_LEN,
+            "Attributes too long, cannot copy into static buffer"
         );
 
+        let mut signature_buffer = [0u8; HASH_DATA_BUFFER_LEN];
+
+        let signature_buffer_slice = &mut signature_buffer[..sig_len];
+        let load_result = signature.load_slice(0, signature_buffer_slice);
+        require!(load_result.is_ok(), "Failed to load attributes into buffer");
```

Comment:
Misleading error message
```suggestion
        require!(load_result.is_ok(), "Failed to load signature into buffer");
```